roachprod: automatically create admin user for secure clusters

renatolabs opened the pull request:
This commit adds a default admin user to clusters created with roachprod; this applies both to the system interface, as well as virtual clusters (shared or separate process).

This is mostly done for convenience, so that roachprod users can have easy access to the dbconsole. It is also helpful while running roachtests.

Epic: none

Release note: None

Commit:
roachprod: automatically create admin user for secure clusters

This commit adds a default admin user to clusters created with
roachprod; this applies both to the system interface, as well as
virtual clusters (shared or separate process).

This is mostly done for convenience, so that roachprod users can have
easy access to the dbconsole. It is also helpful while running
roachtests.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/112765)
<!-- Reviewable:end -->

Commit:
roachprod: support `--secure` mode in `stop-sql`

Previously, shared-process virtual clusters serviced by secure
clusters would not be able to be stopped using `stop-sql`. This was
made worse by the fact that the error checking was wrong, and the
error returned by the `cockroach sql` call was being ignored.

Epic: none

Release note: None

msbutler commented on pkg/roachprod/install/cockroach.go:
```diff
@@ -937,6 +952,62 @@ func (c *SyncedCluster) initializeCluster(
+	}
+
+	const username = "roach"
+	// N.B.: we cannot reuse username and password combinations across
```
nit: we _can_ use the same username and password, but it leads to weird behavior, right?

renatolabs replied:
Reworded the comment here. We _can_, but then the virtual cluster console becomes impossible to access.

msbutler reviewed (approved):
LGTM, modulo some manual testing of roachtests that use roachprod with and without tenants

msbutler commented on pkg/roachprod/install/cockroach.go:
```diff
@@ -937,6 +952,62 @@ func (c *SyncedCluster) initializeCluster(
+		result = results[0]
+		return nil
+	}); err != nil {
+		return &RunResultDetails{}, fmt.Errorf("failed to create default admin user: %w", err)
```
given all the weird and unexpected errors encountered while adding the automatic backup schedule to roachprod, I wonder if it's worth merely logging this error.

renatolabs replied:
I'll keep as-is until we know it's a problem. Since this is only running for secure clusters, the impact is smaller even if this code were completely broken.

renatolabs commented:
> modulo some manual testing of roachtests

Ran an arbitrary c2c roachtest on this branch (`c2c/tpcc/warehouses=500/duration=10/cutover=0`), and it worked fine:

https://teamcity.cockroachdb.com/viewLog.html?buildId=12315069&buildTypeId=Cockroach_Nightlies_RoachtestNightlyGceBazel&tab=buildResultsDiv&branch_Cockroach_Nightlies=112765

Also ran a few other roachtests (that use and don't use secure mode) on my gceworker; no issues.

TFTR!

bors r=msbutler
